src/dex/clear/clear.ts

```diff
@@ -0,0 +1,437 @@
+import { ClearConfig, Adapters } from './config';
+import { ClearData, DexParams } from './types';
+import { Network, NULL_ADDRESS, SwapSide } from '../../constants';
+import * as CALLDATA_GAS_COST from '../../calldata-gas-cost';
+import { getDexKeysWithNetwork } from '../../utils';
+import clearSwapAbi from '../../abi/clear/ClearSwap.json';
+import clearFactoryAbi from '../../abi/clear/ClearFactory.json';
+import { NumberAsString } from '@paraswap/core';
+
+const CLEAR_GAS_COST = 150_000;
+
+interface ClearVault {
+  id: string;
+  address: string;
+  tokens: ClearVaultToken[];
+}
+
+interface ClearVaultToken {
+  id: string;
+  address: string;
+  symbol: string;
+  decimals: string;
+}
+
+/**
+ * Clear DEX Integration for ParaSwap
+ *
+ * Clear is a custom multi-token vault protocol (NOT a standard AMM)
+ * - Vaults contain N tokens (not just 2 like Uniswap pools)
+ * - Pricing via ClearSwap.previewSwap() RPC call (no x*y=k formula)
+ * - Discovery via GraphQL indexer
+ * - Each (vault, tokenA, tokenB) combination = one "pool" for ParaSwap
+ */
+export class Clear extends SimpleExchange implements IDex<ClearData> {
+  static dexKeys = ['clear'];
+  logger: Logger;
+  protected config: DexParams;
+  protected adapters: { [side: string]: { name: string; index: number }[] };
+
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly isFeeOnTransferSupported = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(ClearConfig);
+
+  readonly clearSwapIface = new Interface(clearSwapAbi);
+  readonly clearFactoryIface = new Interface(clearFactoryAbi);
+
+  // Cache for vaults to reduce GraphQL calls
+  private vaultsCache: Map<string, ClearVault[]> = new Map();
+  private vaultsCacheTimestamp: Map<string, number> = new Map();
+  private readonly CACHE_TTL = 60 * 1000; // 1 minute
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    adapters?: { [side: string]: { name: string; index: number }[] },
+    logger?: Logger,
+  ) {
+    super(dexHelper, dexKey);
+    this.config = ClearConfig[dexKey][network];
+    this.adapters = adapters || Adapters[network] || {};
+    this.logger = logger || dexHelper.getLogger(dexKey);
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  /**
+   * Query GraphQL endpoint for data
+   */
+  private async queryGraphQL(query: string): Promise<any> {
+    if (!this.config?.subgraphURL) {
+      throw new Error(`GraphQL endpoint not configured for ${this.dexKey} on ${this.network}`);
+    }
+
+    try {
+      const response = await fetch(this.config.subgraphURL, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ query }),
+      });
+
+      const json: any = await response.json();
```

Comment:
What if the api is down (returning 404 or 500 or whatever) ?

We need to check resp status before decoding it